keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/jackson/DefaultSecurityGroupRuleDeserializer.kt

```diff
@@ -5,12 +5,10 @@ import com.netflix.spinnaker.keel.api.ec2.CrossAccountReferenceRule
 import com.netflix.spinnaker.keel.api.ec2.ReferenceRule
 import com.netflix.spinnaker.keel.api.ec2.SecurityGroupRule
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.boot.jackson.JsonComponent
 
 @JsonComponent
-//@ConditionalOnMissingBean(SecurityGroupRuleDeserializer::class)
-@ConditionalOnProperty("keel.plugins.ec2.defaultSerializers", matchIfMissing = true)
+@ConditionalOnMissingBean(name = ["securityGroupRuleDeserializer"])
```

Comment:
Bonus: I found a simpler way to toggle between the internal and default implementation via the bean name.